src/main/java/nl/igorski/example/MWEngineActivity.java

```diff
@@ -509,6 +475,20 @@ public void handleNotification( int aNotificationId, int aNotificationValue )
                     Log.d( LOG_TAG, "seq. position: " + sequencerPosition + ", buffer offset: " + aNotificationValue +
                             ", elapsed samples: " + elapsedSamples );
                     break;
+
+
+                case RECORDED_SNIPPET_READY:
+                    runOnUiThread( new Runnable() {
+                        public void run() {
```

Comment:
By making the writing of an in-memory recording to storage an asynchronous operation, we can more easily run this process in a separate thread. This prevents the stuttering of buffer under runs occurring while recording long segments while the sequencer is running.